### DISCUSSION TITLE
How to get the google calendar events in Maui

### DISCUSSION THREAD

**Discussion by Sasikumar3595** [dotnet/maui#3847] [Q&A]
I was creating the sample to get all the google calendar events and show the events with details on list. But it always throws the exception while fetching the data.

I tried to get events by follow the below link
https://www.c-sharpcorner.com/article/get-and-create-google-calendar-events-from-net/

and i also refer the google link also 
https://developers.google.com/api-client-library/dotnet/get_started#simple

The issue was raised on make a request and i tried with both API key and client service.
Events events = request.Execute();

I created the calendar credential and request by below

```
GoogleCredential credential;  
 using(var stream = new FileStream(keyfilepath, FileMode.Open, FileAccess.Read)) {     
       credential = GoogleCredential.FromStream(stream)  
        .CreateScoped(Scopes).CreateWithUser("demo123@gmail.com");  
 }  

var service = new CalendarService(new BaseClientService.Initializer() {  
       HttpClientInitializer = credential,  
        ApplicationName = "Calendar Sample",  
 }); 

 EventsResource.ListRequest request = service.Events.List("primary"); 
 Events events = request.Execute(); /// Makes crash
```

**Android Crash** : System.Net.WebException: 'Unable to resolve host "oauth2.googleapis.com": No address associated with hostname'

**Windows** : Google.Apis.Auth.OAuth2.Responses.TokenResponseException: 'Error:"unauthorized_client", Description:"Client is unauthorized to retrieve access tokens using this method, or client not authorized for any of the scopes requested.", Uri:""'

Suggest me to resolve the problem.

**Comment by nogginbox** [comment]
What type of exception and what was the exception's message?

This doesn't sound like a MAUI specific thing. Depending on what the exception is, you might get more people looking at your question and more chance of a helpful answer on Stackoverflow.

**Reply by Sasikumar3595** [reply]
@NogginBox 

I created the calendar credential and request by below

```
GoogleCredential credential;  
 using(var stream = new FileStream(keyfilepath, FileMode.Open, FileAccess.Read)) {     
       credential = GoogleCredential.FromStream(stream)  
        .CreateScoped(Scopes).CreateWithUser("demo123@gmail.com");  
 }  

var service = new CalendarService(new BaseClientService.Initializer() {  
       HttpClientInitializer = credential,  
        ApplicationName = "Calendar Sample",  
 }); 

 EventsResource.ListRequest request = service.Events.List("primary"); 
 Events events = request.Execute(); /// Makes crash
```

**Android Crash** : System.Net.WebException: 'Unable to resolve host "oauth2.googleapis.com": No address associated with hostname'

**Windows** : Google.Apis.Auth.OAuth2.Responses.TokenResponseException: 'Error:"unauthorized_client", Description:"Client is unauthorized to retrieve access tokens using this method, or client not authorized for any of the scopes requested.", Uri:""'

**Comment by saiganesh-sakthivel** [comment]
any update @NogginBox ???

**Reply by nogginbox** [reply]
I don't work for Microsoft. I'm a user of MAUI and not a maintainer or developer working on the project. I just try to help sometimes if I can by answering questions or pointing people towards resources that might help them.

Did you ask the question on [Stackoverflow](https://stackoverflow.com/) like I suggested?

I have never used the google calendar library so I can't help with that.

The only MAUI specific thing I can think of, is setting up internet connectivity (but that would only apply to Android and IoS)

* [MAUI Connectivity](https://learn.microsoft.com/en-us/dotnet/maui/platform-integration/communication/networking?tabs=android)
